src/server/status_server.rs

```diff
@@ -48,19 +59,27 @@ impl StatusServer {
         let builder = Server::try_bind(&addr)?;
 
         // Create a status service.
-        let service = |req: Request<Body>| -> FutureResult<Response<Body>, hyper::Error> {
-            let mut response = Response::new(Body::empty());
+        let service = |req: Request<Body>| -> Box<dyn Future<Item=Response<Body>, Error=hyper::Error> + Send> {
+            let path = req.uri().path().to_owned();
+            let method = req.method().to_owned();
 
-            match (req.method(), req.uri().path()) {
-                (&Method::GET, "/metrics") => {
-                    *response.body_mut() = Body::from(dump());
+            #[cfg(not(feature = "no-fail"))]
+                {
+                    if path.starts_with(FAIL_POINTS_REQUEST_PATH) {
```

Comment:
This looks complex. Maybe keep matching but do noop if feature is not enabled can simplify the code?

Reply:
what is noop? I am not sure I understand your input here :P

Basically the logic here is, if feature fail-point is on, then use `handle_fail_points_request()` to deal with request that has path starting with "fail"

Reply:
I mean we can use a syntax like the other one, but do nothing when it is matched but feature is not enabled.

Reply:
还是没太听明白啥意思... :P "the other one" 是哪个?

Reply:
I think what he mean is to remove feature guard here and write two different implements of function `handle_fail_points_request` for different feature configurations.

Reply:
If the user doesn't have failpoints enabled it might be good from a user-learning perspective to have it report back to them a suggestion how to turn it on.

Reply:
Good point, I will see whether I can add some output later. Do you remember how to turn it on while build? I think I forgot about it now... :P